mula/scheduler/schedulers/boefje.py

```diff
@@ -413,6 +414,17 @@ def is_task_running(self, task: BoefjeTask) -> bool:
             )
             raise exc_db
 
+        if (
+            task_db is not None
+            and task_db.status == TaskStatus.DISPATCHED
+            and (
+                task_db.modified_at is not None
+                and datetime.now(timezone.utc)
+                > task_db.modified_at + timedelta(seconds=self.ctx.config.pq_grace_period)
+            )
+        ):
+            raise TaskStalledError()
```

Comment:
If find this confusing. The function is changed to also be able to return whether a task is stalled, but instead of changing the function signature to reflect that it can return three different states an exception is thrown.

Reply:
Im ok with this change. The task is running or not, that seems to make sense. A stalled task is however an unexpected situation, and as such warrants an exception in my book.

Reply:
It is just one of the things the scheduler needs to handle in the normal course of doing its job. It is not an unexpected situation like failing to connect to something, inconsistent state or a connection that gets reset.

In my opinion you create the most maintainable code over the long run when methods work how someone new to the code would expect them to work. If such a person would have to write some new code where there is a need to check if a task is still running, then the expectation would in my eyes be that `is_task_running` returns just that and not also return other states via exceptions that would have to be explicitely handled. And then if you don't handle the exception everything just works even in QA, but then you deploy in production and it fails after some time when it encouters a stalled task.

In my opinion the goal should not be that we just write code that works, but also code that can still be maintained years in the future, when we either have forgotten how we implemented things years earlier, or maybe we have moved on and other people are working on the project.